packages/shared/src/components/checklist/NotificationChecklistStep.tsx

```diff
@@ -1,15 +1,23 @@
-import React, { ReactElement } from 'react';
+import React, { ReactElement, useContext } from 'react';
 import { ChecklistStepProps } from '../../lib/checklist';
 import { Button } from '../buttons/Button';
 import { ChecklistStep } from './ChecklistStep';
 import BellIcon from '../icons/Bell';
+import NotificationsContext from '../../contexts/NotificationsContext';
+import { NotificationPromptSource } from '../../lib/analytics';
 
 const NotificationChecklistStep = (props: ChecklistStepProps): ReactElement => {
-  // TODO WT-1293-checklist-components add business logic
+  const { onTogglePermission } = useContext(NotificationsContext);
 
   return (
     <ChecklistStep {...props}>
-      <Button icon={<BellIcon />} className="btn-primary">
+      <Button
+        icon={<BellIcon />}
+        className="btn-primary"
+        onClick={() => {
+          onTogglePermission(NotificationPromptSource.SquadPage);
```

Comment:
But this won't save it in the database right?
I think we need that part for other actions we might trigger later.

Reply:
It will get triggered from `useEffect` in `NotificationsContextProvider` if user actually accepts the prompt from browser to allow notifications. 

We can call `completeAction` here to complete the step no matter if user actually accepts notifications or not - this is a question from my point of view.

Reply:
Ah no was just missing that context in this PR, if that's the case it's fine with me.
Maybe just ask @sshanzel if that was his intended behaviour.

Reply:
If our goal is to just make user click the CTA, we can probably send the completeAction already, but if we want to force them to subscribe before completing, then we will have to rely on the context's capacity. Probably a question for product on their intent with this one.

Reply:
From [spec](https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/472743940/Squads+closed+beta+v5.0) it says:
<img width="264" alt="image" src="https://github.com/dailydotdev/apps/assets/9803078/f2112b53-cbe8-4189-8287-c3b9fc48b452">

Current implementation matches that so I would leave it at this.

Reply:
I actually think the discussions mention click not completion, we had this discussion already.